### DISCUSSION TITLE
unable to run in browser environemnt

### DISCUSSION THREAD

**Discussion by techy-ashir** [prisma/prisma#22998] [Q&A]
### Bug description

Error submitting form: Error: PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in `unknown`).

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```

```

**Comment by PandiyanCool** [comment]
Facing similar issue.

**Comment by ludralph** [comment]
Hi @techy-ashir 👋 

The error message "PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in unknown)" typically occurs when you're trying to run Prisma Client in a browser environment. 

Prisma Client is designed to run in a server-side environment and does not work in the browser.

If you're using Next.js, there are specific ways to ensure Prisma runs on the server-side:

- If you're using the Next.js app router, add "use server" at the top of your file.
- If you're using the Next.js pages router, keep the files that import Prisma or contain any Prisma query function inside the /api directory.

This is because Prisma does not work on the client side as the browser does not execute SQL queries. Prisma has to be added with your Node.js app or inside the backend functions of Next.js.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by jrmyio** [comment]
I got this issue since 5.11.0. I am using vite to build a node js application.

Since 5.11.0 it doesnt seem to work anymore.

**Comment by vjpr** [comment]
_.prisma/client/default.js_

```
module.exports = { ...require('.') }
```

_.prisma/client/package.json_

```
{
  "name": "prisma-client-5e008f5f00b7dadeea7ddcf9071f1dee6b368f07f553695da13bc7f51549731c",
  "main": "index.js",
  "types": "index.d.ts",
  "browser": "index-browser.js",
  "exports": {
    "./package.json": "./package.json",
    ".": {
      "require": {
        "node": "./index.js",
        "edge-light": "./wasm.js",
        "workerd": "./wasm.js",
        "worker": "./wasm.js",
        "browser": "./index-browser.js",
        "default": "./index.js"
      },
      "import": {
        "node": "./index.js",
        "edge-light": "./wasm.js",
        "workerd": "./wasm.js",
        "worker": "./wasm.js",
        "browser": "./index-browser.js",
        "default": "./index.js"
      },
      "default": "./index.js"
    },
    "./edge": {
      "types": "./edge.d.ts",
      "require": "./edge.js",
      "import": "./edge.js",
      "default": "./edge.js"
    },
    "./react-native": {
      "types": "./react-native.d.ts",
      "require": "./react-native.js",
      "import": "./react-native.js",
      "default": "./react-native.js"
    },
    "./extension": {
      "types": "./extension.d.ts",
      "require": "./extension.js",
      "import": "./extension.js",
      "default": "./extension.js"
    },
    "./index-browser": {
      "types": "./index.d.ts",
      "require": "./index-browser.js",
      "import": "./index-browser.js",
      "default": "./index-browser.js"
    },
    "./index": {
      "types": "./index.d.ts",
      "require": "./index.js",
      "import": "./index.js",
      "default": "./index.js"
    },
    "./wasm": {
      "types": "./wasm.d.ts",
      "require": "./wasm.js",
      "import": "./wasm.js",
      "default": "./wasm.js"
    },
    "./runtime/library": {
      "types": "./runtime/library.d.ts",
      "require": "./runtime/library.js",
      "import": "./runtime/library.js",
      "default": "./runtime/library.js"
    },
    "./runtime/binary": {
      "types": "./runtime/binary.d.ts",
      "require": "./runtime/binary.js",
      "import": "./runtime/binary.js",
      "default": "./runtime/binary.js"
    },
    "./generator-build": {
      "require": "./generator-build/index.js",
      "import": "./generator-build/index.js",
      "default": "./generator-build/index.js"
    },
    "./sql": {
      "require": {
        "types": "./sql.d.ts",
        "node": "./sql.js",
        "default": "./sql.js"
      },
      "import": {
        "types": "./sql.d.ts",
        "node": "./sql.mjs",
        "default": "./sql.mjs"
      },
      "default": "./sql.js"
    },
    "./*": "./*"
  },
  "version": "5.20.0",
  "sideEffects": false
}
```

It seems like vite is building the browser version. But for the SSR or Vercel function bundle, you want the node version or worker version of Prisma.

> The default allowed conditions are: import, module, browser, default, and production/development based on current mode
>
> See: https://v3.vitejs.dev/config/shared-options.html#resolve-conditions

It looks like there is no `node` condition in this list that vite looks for. So maybe it ends up with the `browser` condition.

_node_modules/@prisma/client_

```
  "exports": {
    ".": {
      "require": {
        "types": "./default.d.ts",
        "node": "./default.js",
        "edge-light": "./default.js",
        "workerd": "./default.js",
        "worker": "./default.js",
        "browser": "./index-browser.js"
      },
      "import": {
        "types": "./default.d.ts",
        "node": "./default.js",
        "edge-light": "./default.js",
        "workerd": "./default.js",
        "worker": "./default.js",
        "browser": "./index-browser.js"
      },
      "default": "./default.js"
    },
```

I added `resolve.conditions = ['node']` which adds node as an additional condition to look for.

---

You can also use `resolve.alias` to redirect `.prisma/client/default: ./node_modules/.prisma/client/index`

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
